app/models/miq_worker/runner.rb

```diff
@@ -386,7 +385,8 @@ def heartbeat_to_drb
   end
 
   def heartbeat_to_file
-    FileUtils.touch(@worker.heartbeat_file)
+    timeout = worker_settings[:heartbeat_timeout] || Workers::MiqDefaults.heartbeat_timeout
```

Comment:
Since we're going to have the worker write his timeout to the heartbeat file, what do you think about including [this existing logic](https://github.com/ManageIQ/manageiq/blob/8b04e1335c09ea314d61859446356f5e9cc38cc0/app/models/miq_server/worker_management/monitor/settings.rb#L30-L44) where the timeout of the an active queue message (for queue workers only) is the timeout if the worker is processing the message? I think we'll still need to support that in some form in the new architecture.

With that said, I'm not sure how that can be done because the worker heartbeats before taking work from the queue.